soccer/src/soccer/strategy/agent/position/waller.cpp

```diff
@@ -47,12 +47,16 @@ std::optional<RobotIntent> Waller::get_task(RobotIntent intent, const WorldState
 
     if (abs(robot_pos.dist_to(goal_pos) - min_wall_rad) < kRobotRadius &&
         robot_pos.dist_to(target_point) > kRobotRadius) {
-        uint8_t parent_id;
+        uint8_t parent_id =
```

Comment:
mind following through with your comment and taking this back out?
if you don't like the declaration and then assignment in the if-else (and there are good arguments against it)
one nice c++ pattern is an immediately invoked lambda: 

something like this:
```
uint8_t robot_id = [&](){
   if (...) return walling_robots_[waller_pos_ - 2];
   else return robot_id;
}();
```

Reply:
this forces the programmer to always return something so it's guaranteed robot_id will always have a value, but it can be based on a complex conditional.

Reply:
you can also just leave it with the previous pattern tho